code/modules/awaymissions/corpse.dm

```diff
@@ -96,8 +102,8 @@
 		if(ishuman(M))
 			var/mob/living/carbon/human/hoomie = M
 			hoomie.body_type = mob_gender
-	if(faction)
-		M.faction = list(faction)
+	if(LAZYLEN(faction))
+		M.faction = faction.Copy()
```

Comment:
Does this faction ever change?
If not you could use the `string_list()` proc to make it a string list.
Those are cached by their contents, reducing the amount of duplicate lists.
They can be set on intialization, and then you can simply do `M.faction = faction`
And if you want to add or remove factions you do something like this:
`faction = string_list(faction - removed_values)`
`faction = string_list(faction | or_added_values)`

Reply:
Gave it a shot, have a look.